portfolio/src/main/webapp/script.js

```diff
@@ -76,16 +76,42 @@ function updateComments(comments, element) {
 
 /** Creates an <li> element containing text. */
 function createListElement(comment) {
-    console.log('formatted: ' + comment.formatted_time);
     const liElement = document.createElement('li');
+    var rounded_score = comment.sentiment_score.toFixed(2);
+
     liElement.innerHTML = comment.content + '<br><br><div class="comment-data">Posted by ' + 
         comment.user_name + ' (' + comment.user_location + ') at ' + comment.formatted_time +
-        '</div>';
-    console.log('innerHTML: ' + liElement.innerHTML);
+        '<br>Sentiment score: ' + rounded_score + '</div>';
+
     liElement.setAttribute("class", "comment");
+    setSentimentColor(liElement, comment.sentiment_score);
     return liElement;
 }
 
+/** Set background color based on sentiment score */
+function setSentimentColor(liElement, score) {
+    console.log('score:' + score);
+    switch(true) {
+        case (score < -0.3):
+            // Negative - red
+            console.log('negative');
+            liElement.style.backgroundColor = "#e8dada";
+            break;
+        case (score < 0.3):
+            // Neutral - blue
+            console.log('neutral');
+            liElement.style.backgroundColor = "#dcdae8";
+            break;
+        case (score <= 1):
+            // Positive - green
+            console.log('positive');
+            liElement.style.backgroundColor = "#dae8da";
+            break;
+        default:
```

Comment:
This is not a request for a change, but just some background on alternate color spaces.

At the risk of sending you down a rabbit hole, I wanted to point out something else cool that can be done with colors. Especially if you use a different color space than RGB. In particular, if you use HSL (Hue, Saturation, Lightness), you can get an easy way to vary how light a color is by just adjusting one parameter -- the lightness.

So you could pick a general hue like red, blue or green and then adjust the lightness / darkness (probably in the range of 50%-85/90% based on the actual score. Something perhaps like
```
case (score < -0.3):
    // Set lightness to the range of 50% - 85%, as score will range from -1 to -0.3 => -50 to -15.
    liElement.style.backgroundColor = "hsl(0, 100%, " + (100 + Math.round(score * 50) + "%);
```

CSS support HSL color via the hsl() function that can be used as a color value.
See https://www.w3schools.com/colors/colors_hsl.asp

But only do this if you have some time left over.

Reply:
Thanks for this tip! I will test it out.